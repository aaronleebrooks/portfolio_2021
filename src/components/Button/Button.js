import cx from 'classnames';
import styles from './index.module.css';

function Button(props) {
  return (
    <button className={cx(styles.button, props.className)} onClick={props.onClick}>
      {props.children}
    </button>
  );
}

export default Button;
