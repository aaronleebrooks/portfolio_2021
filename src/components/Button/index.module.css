.button {
    appearance: none;
    background: none;
    border: none;
}
