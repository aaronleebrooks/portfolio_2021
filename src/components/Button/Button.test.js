import { fireEvent, render } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  const baseProps = {
    onClick: jest.fn(),
    children: "test element"
  };

  test('renders with props', () => {
    const { container } = render(<Button {...baseProps} />);
    expect(container).toMatchSnapshot();
  });

  test('no-op when button is clicked', () => {
    const { getByText } = render(<Button {...baseProps} />);
    fireEvent.click(getByText(baseProps.children));
    expect(baseProps.onClick).toHaveBeenCalledTimes(1);
  });
})
