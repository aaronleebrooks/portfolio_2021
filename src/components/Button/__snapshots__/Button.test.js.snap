// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with props 1`] = `
<div>
  <button
    class="button"
  >
    test element
  </button>
</div>
`;
